sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -160,17 +166,22 @@ case class TakeOrdered(limit: Int, sortOrder: Seq[SortOrder], child: SparkPlan)
 
   private val ord: RowOrdering = new RowOrdering(sortOrder, child.output)
 
-  private def collectData(): Array[InternalRow] =
-    child.execute().map(_.copy()).takeOrdered(limit)(ord)
+  @transient private val projection = projectList.map(newProjection(_, child.output))
```

Comment:
I'm confused here that we have to add `@transient` for it to avoid `NotSerializableException` . It looks to me that `TakeOrderedAndProject` should run on driver side and it doesn't pass its reference to RDD functions. Sorry if I missed something here.